datumaro/plugins/searcher.py

```diff
@@ -0,0 +1,171 @@
+# Copyright (C) 2023 Intel Corporation
+#
+# SPDX-License-Identifier: MIT
+
+import logging as log
+import os
+import os.path as osp
+import urllib
+
+import cv2
+import numpy as np
+from tqdm import tqdm
+from transformers import CLIPTokenizer
+
+from datumaro.plugins.openvino_plugin.launcher import OpenvinoLauncher
+
+
+def download_file(url: str, file_root: str):
+    req = urllib.request.Request(url)
+    with urllib.request.urlopen(req) as source, open(file_root, "wb") as output:  # nosec B310
+        with tqdm(
+            total=int(source.info().get("Content-Length")),
+            ncols=80,
+            unit="iB",
+            unit_scale=True,
+            unit_divisor=1024,
+        ) as loop:
+            while True:
+                buffer = source.read(8192)
+                if not buffer:
+                    break
+
+                output.write(buffer)
+                loop.update(len(buffer))
+    return 0
+
+
+def sigmoid(x):
+    return 1 / (1 + np.exp(-x))
+
+
+def encode_discrete(x):
+    prob = sigmoid(x)
+    z = np.sign(prob - 0.5)
+    return z
+
+
+def normalize(x, axis=1, eps=1e-12):
+    denom = max(np.linalg.norm(x, axis=axis, keepdims=True), eps)
+    return x / denom
+
+
+def compute_hash(features):
+    features = encode_discrete(features)
+    features = normalize(features, axis=-1)
+    hash_key = features >= 0
+    hash_key = hash_key * 1
```

Comment:
In addition, those four lines can be 
```
features = np.sign(features, dtype=np.int8)
features = np.clip(features, 0, None)
```

Reply:
I tried to use this like below. I think this result is not same as that I intended due to my answer above.
```
features = features.astype(np.int8)
features = np.sign(features)
features = np.clip(features, 0, None)
```

Reply:
I updated this part as your suggestion. I removed `normalize` function.